test/interactive_tarantool.lua

```diff
@@ -138,11 +159,18 @@ end
 -- * ReadLine 7: x<CR><duplicate x>, extra spaces.
 -- * ReadLine 6: <space><CR>.
 --
--- This function drop duplicates that goes in row, strips <CR> and
--- spaces. Applying of this function to a source command and
--- readline's echoed command allows to compare them for equality.
+-- This function drop duplicates that goes in row, strips <CR>,
+-- spaces, tabs and ANSI color codes. Applying of this function
+-- to a source command and readline's echoed command allows to
+-- compare them for equality.
 local function _prepare_command_for_compare(x)
-    x = x:gsub(' ', ''):gsub(M.CR, '')
+    -- The terrifying regexp in the last gsub is meant to clean ANSI
+    -- color codes, for more details on the format see the doc:
+    -- https://gist.github.com/fnky/458719343aabd01cfb17a3a4f7296797
+    x = x:gsub(' ', '')
+         :gsub(M.CR, '')
+         :gsub(M.TAB, '')
+         :gsub('[\x1b\x9b][][()#;?%d]*[A-PRZcf-ntqry=><~]', '')
```

Comment:
If it is just about colors, it should be much simpler. According to the link something like `:gsub(M.ESC .. '%[[0-9;]*m', '')` should work. If this is not only about the colors, the comment seems misleading.

Reply:
BTW, I would put such a regexp into the exposed module constants (close to M.EXP and others). It can be useful outside of the module itself.

Reply:
Good idea! Moved the `:gsub()` to `M.DECOLOR_RE`.

Reply:
It looks much more accurate now.

Nit: I'm OK with regexp `M.DECOLOR_RE` or function `M.decolor`. But, please name/place it in sync with other functions/constants.

Reply:
Oh, my bad, will fix.

Reply:
Fixed.